### ISSUE TITLE
documentation of ... anything?

### ISSUE THREAD

**Issue by 0x0lli** [KeepThisTicket/weatherClock#6]
after cloning the repo and configuring the display into the correct rotation, I am left with questions only.

1. how _are_ you running the file itself? If I run it over SSH via "python weatherClock.py" I get:
  File "weatherClock.py", line 326
SyntaxError: Non-ASCII character '\xe2' in file weatherClock.py on line 326, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details"
 Cooperation of b-morgan and Vacant0mens produces a syntax error within their v4 in line 22. Trying to run the file on the GUI brings the source to an editor. It appears, some of you have it running, but how?

2. My intention would be to have it running when powered on, so how would one go about it? I found there are several ways, just trying to see if your can recommend one.  https://www.dexterindustries.com/howto/run-a-program-on-your-raspberry-pi-at-startup/
3. the pimoroni link is not working anymore.

**Comment by 0x0lli** [comment]
after having set the locale several times unsuccessfull, I fixed the first error via inserting in line 1:
# coding: utf-8

I run the .py via "python weartherClock.py -a [API]"
now this is the new error:

###
Traceback (most recent call last):
  File "weatherClock.py", line 9, in <module>
    from pynput.mouse import Listener
  File "/usr/local/lib/python2.7/dist-packages/pynput/__init__.py", line 40, in <module>
    from . import keyboard
  File "/usr/local/lib/python2.7/dist-packages/pynput/keyboard/__init__.py", line 31, in <module>
    backend = backend(__name__)
  File "/usr/local/lib/python2.7/dist-packages/pynput/_util/__init__.py", line 82, in backend
    if resolutions else '')
ImportError: this platform is not supported: ('failed to acquire X connection: Bad display name ""', DisplayNameError())

Try one of the following resolutions:

 * Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly
###

then, I set putty to forward X11 to localhost:0.0 #not working# 
then to the pi's IP instead, #not working# 

The thing is, I installed rpiOS full (without the extra software but not the lite version) on a rpi0W, updated the packages, installed the display via the installer script off of their site ("curl https://get.pimoroni.com/hyperpixel4 | bash") as the link to the recommended instructions is wrong (it appears, the display isn't too well fit for the pi4 and they took the instructions down). 
When installing the display, one needs to specify the year of production, which is not possible as there is no value shown or at least I dont know where to look for it. As the display just arrived at my door from digikey, I assumed it is the latest model, therefore chose option "Weirdly Square - Pi 3B+ or older" and production date as 2021 or newer. #not working# changed to other manufactoring date (2020) #not working#

What is it with this error?

**Comment by 0x0lli** [comment]
solved this problem with 
"export DISPLAY=:0.0" on shell just before running the .py

now, I get this: 

{u'message': u'Invalid API key. Please see http://openweathermap.org/faq#error401 for more info.', u'cod': 401}

... Main Loop Start ...

3 43 6

---- updateForecast() ----
hourCursor: 3

current hour: 15 PM
forecast hour: 15
Traceback (most recent call last):
  File "weatherClock.py", line 451, in <module>
    updateForecast()
  File "weatherClock.py", line 260, in updateForecast
    print("temperature: " + str(data["hourly"][num]["temp"]))
KeyError: 'hourly'



Setup up another API key, nothing changes, error is the same. I read, it takes some time for the API key to be activated. Will try tomorrow.
Is there a need to have a paid subscription for the openweathermap or select one particular subscription? I thought there was only one option in free plan.

**Comment by flyerfanatic** [comment]
0x0lli I don't have anything to add to your comments at this point but I wanted to say thank you for helping out. I have the same issue with the display. I am a raspberry pi noob so all comments are helpful to me.

I have been using SSH and testing the code with X11 forwarding via MobaXterm. The odd thing is that when I ran python3 weatherClock.py the clock would pop up on my windows 10 machine and the hyperpixel display would only have the raspianOS desktop.

When I get home from work today I will try running "export DISPLAY=:0.0" before python3 weatherClock.py.

**Comment by flyerfanatic** [comment]
I finally got it to display on my hyperpixel! my fix was to use `export DISPLAY=:0` (NOT 0.0)

for anyone curious you can tell if you have an export issue with `echo $DISPLAY` if that returns a blank line then you need to export your display.

Now, how to figure out how to run the clock in fullscreen is the next task. currently it has the taskbar at the top.

**Comment by flyerfanatic** [comment]
while working on getting the program to run on boot I ran into a few problems. I don't want to have to use `export DISPLAY=:0` every time I reboot the pi. so I did some googling and found that adding `export DISPLAY=:0` to the end of the .bashrc file allows me to set it on boot.

to do this simply enter the following into your terminal:

`sudo nano /home/pi/.bashrc`

then add `export DISPLAY=:0`

then ctrl+x then y then enter to save

**Comment by flyerfanatic** [comment]
full screen update here.

in the wn.setup line use the following to cover the task bar: `wn.setup(width=720, height=720, startx=-1, starty=-1)`

there will be a border around the right and bottom sides but that's the best I have figured out so far.

**Comment by KeepThisTicket** [comment]
Thanks for all your guys' efforts and contributions so far. Apologies for not replying quicker, but unforunately I did not have the same issues that you guys were having. It could definitely be that I forgot to document some things when I first started this project and those minor things became big things when the project became public - sorry for that ! 

I'll be going through the other issues and PRs to hopefully resolve as many outstanding ones as I can

Regarding points 1 and 2, i never really figured how to get the program to run at boot, so I just turn it on and double tap the .py script to open the program and then I leave it running.

**Comment by 0x0lli** [comment]
@KeepThisTicket : nice to see you got back. Thanks for this cool idea. I think it makes for the most indiviual present for a dog owner apart from the decorative effect.

@flyerfanatic fantastic find. In turtle description, I found there is no parameter on where to stop the turtle, only start pixels for x and y as given in your post. I tweaked the numbers a bit but your parameteres are the best one I could find. Question still remains: how to get rid of the last lines of distraction? BTW check out this contribution, it is functional with the icons (a first for me), only the area displayed is a little smaller than the official rls. https://github.com/KeepThisTicket/weatherClock/pull/9/files

**Comment by KeepThisTicket** [comment]
Closing this issue for now, as issues are now resolved with PR#8 merge.

PR#9 which you mentioned above shows the code needed to auto-run on startup, but not tested as of yet. 

Try pulling the latest code and seeing if that works for you, as it works fine for me as well as others. If it doesn't, please open another issue with the issues you are still seeing !

**Comment by 0x0lli** [comment]
although closed, I am taking it as a temporary notepad:

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@&%/////////////%&@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@&##( (##&@@@#///////////////////#@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@#             ,//////////////////////%@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@&,                */////////////////////(@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@                   *(/*,*////////////////%@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@                            *////////////%@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@                                  ,///////////%@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@&                                     *//////////#@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@,                                           .*//(@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@(                                               #@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@(                                               (@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@%                                             ,&@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@(                                          %@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
''' 

here's what I've done on my windows machine, 2021-09-26, to get it working on a "raspberry pi zero WH" connected to a "hyperpixel 4 weirdly square"

-1. have the case printed, get yourself a powersupply with a hardware 0/1-switch

0a. ```https://home.openweathermap.org/users/sign_up``` apply for your API key
0b. use google maps to find out your geolocation, ie: langitude and longitude values

1a. flash a SD card via "balea etcher" with "Raspberry Pi OS with desktop" // Release date: May 7th 2021 // Kernel version: 5.10 // Size: 1,180MB
1b. after flashing, disconnect SD Card, connect again, copy both "SSH" and (a configured) "wpa_supplicant.conf" to the /boot/-partition (then, soft-disconnect the SD before unplugging)
1c. insert SD into pi, power on pi
1d. (for windows:) install "putty" and "VNC Viewer" on your machine 
1e. connect to the pi via putty - for first time login you need: "login:pi / pass:raspberry" (needed IP of the pi can be found via tools such as "Net Analyzer" or "PortDroid", for android, that is) 

2. ```sudo raspi-config``` set the following: {system options:password,hostname // interface options:VNC // localisation:timezone}, reboot

3. ```sudo apt update; sudo apt full-upgrade -y; pip3 install pynput; git clone https://github.com/KeepThisTicket/weatherClock; curl https://get.pimoroni.com/hyperpixel4 | bash```  # will take some time, once you're in the pimoroni hyperpixel dialogue, my answers are: [y - 3 - 0 - y]

4. ```sudo nano /home/pi/weatherClock/settings.json``` edit your settings, enter API, lat, long, theme, title etc.

5. ```sudo nano /home/pi/.bashrc```
add those lines to the end of the file:
```export DISPLAY=:0
cd /home/pi/weatherClock/
python3 weatherClock.py```

6. ```sudo sed -i -- "s/#xserver-command=X/xserver-command=X -nocursor/" /etc/lightdm/lightdm.conf```   # removes the mouse cursor icon for touchscreen

7. ```sudo nano /boot/config.txt```
replace ```dtoverlay=hyperpixel4``` with ```dtoverlay=hyperpixel4:rotate``` and add the following line direclty thereafter:
```display_rotate=0```

8. reboot.

9. connect via VNC and finish the welcome dialogue (might be redundant but I did not figure how to get rid of that GUI dialogue without finishing it on the GUI)

10. reboot, now everything should power-up automatically.
